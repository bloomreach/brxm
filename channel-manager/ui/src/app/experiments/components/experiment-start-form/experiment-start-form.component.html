<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<ng-container *ngIf="selectedVariant">
  <em-experiment-info-bar *ngIf="againstDefault">
    {{ 'EXPERIMENT_INFO_START_AGAINST_DEFAULT_GOAL_REQUIRED' | translate:{ variantName: selectedVariant?.name } }}
  </em-experiment-info-bar>
  <em-experiment-info-bar *ngIf="!againstDefault">
    {{ 'EXPERIMENT_INFO_START_ALL_VARIANTS_GOAL_REQUIRED' | translate:{ variantName: getOriginalVariant(selectedVariant)?.name } }}
  </em-experiment-info-bar>
</ng-container>

<div class="form-wrapper">
  <mat-form-field appearance="outline">
    <mat-label class="qa-variant-select-label">
      {{ 'VARIANTS' | translate }}
    </mat-label>
    <mat-select class="qa-variants-select"
                [(value)]="selectedVariant"
                [disabled]="disabled">
      <mat-option *ngFor="let variant of groupedVariants"
                  class="{{ 'qa-variant-select-option-' + variant.id }}"
                  [value]="variant">
        {{ variant.name }}
      </mat-option>
    </mat-select>
  </mat-form-field>

  <mat-form-field appearance="outline">
    <mat-label class="qa-goal-select-label">
      {{ 'EXPERIMENT_GOAL' | translate }}
    </mat-label>
    <mat-select class="qa-goal-select"
                [(value)]="selectedGoalId"
                [disabled]="disabled">
      <mat-option><i>{{ 'NONE_OPTION' | translate }}</i></mat-option>
      <mat-option *ngFor="let goal of goals"
                  class="{{ 'qa-goal-select-option-' + goal.id }}"
                  [value]="goal.id">
        {{ goal.name }}
      </mat-option>
    </mat-select>
  </mat-form-field>
</div>

<mat-divider></mat-divider>

<div class="actions">
  <button mat-stroked-button
          [disabled]="isDiscardButtonDisabled || disabled"
          (click)="onDiscard()"
          class="qa-discard">
    {{ 'DISCARD' | translate }}
  </button>
  <button mat-stroked-button
          color="primary"
          [disabled]="isSaveButtonDisabled || disabled"
          (click)="onSave()"
          class="save-button qa-save">
    {{ 'SAVE' | translate }}
  </button>
</div>
