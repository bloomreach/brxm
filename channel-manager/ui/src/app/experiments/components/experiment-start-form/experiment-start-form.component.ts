/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the 'License');
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an 'AS IS' BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';

import { Variant } from '../../../variants/models/variant.model';
import { ExperimentGoal } from '../../models/experiment-goal.model';

interface GroupedVariant {
  ids: string[];
  name: string;
}

@Component({
  selector: 'em-experiment-start-form',
  templateUrl: 'experiment-start-form.component.html',
  styleUrls: ['experiment-start-form.component.scss'],
})
export class ExperimentStartFormComponent implements OnChanges {
  @Input()
  variants: Variant[] = [];

  @Input()
  goals: ExperimentGoal[] = [];

  groupedVariants: GroupedVariant[] = [];
  selectedVariant?: GroupedVariant;
  selectedGoal: ExperimentGoal | undefined;

  get againstDefault(): boolean {
    return this.selectedVariant ? this.selectedVariant.ids.length === 1 : true;
  }

  ngOnChanges(changes: SimpleChanges): void {
    if ('variants' in changes && this.variants) {
      const variantsWithoutDefault = this.filterOutDefault(this.variants);
      this.groupedVariants = this.groupVariants(variantsWithoutDefault);

      if (!this.selectedVariant && this.groupedVariants.length > 0) {
        this.selectedVariant = this.groupedVariants[0];
      }
    }
  }

  private filterOutDefault(variants: Variant[]): Variant[] {
    return variants.filter(v => !v.id.includes('default'));
  }

  private groupVariants(variants: Variant[]): GroupedVariant[] {
    return variants.map(v => ({ ids: [v.id], name: v.variantName }));
  }
}
