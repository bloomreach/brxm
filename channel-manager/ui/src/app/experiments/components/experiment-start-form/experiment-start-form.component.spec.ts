/*
 * Copyright 2020-2023 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NO_ERRORS_SCHEMA, Pipe, PipeTransform, SimpleChange } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatDividerModule } from '@angular/material/divider';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatSelectModule } from '@angular/material/select';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { first } from 'rxjs/operators';

import { Variant } from '../../../variants/models/variant.model';
import { VariantsService } from '../../../variants/services/variants.service';
import { ExperimentGoal } from '../../models/experiment-goal.model';

import { ExperimentStartFormComponent, SelectedVariantIdAndGoalId } from './experiment-start-form.component';

@Pipe({name: 'translate'})
export class TranslateMockPipe implements PipeTransform {
  transform(value: string): string {
    return value;
  }
}

describe('ExperimentStartFormComponent', () => {
  let fixture: ComponentFixture<ExperimentStartFormComponent>;
  let component: ExperimentStartFormComponent;

  const mockVariants = [
    {
      id: 'variant-1',
      variantName: 'Variant-A',
    },
    {
      id: 'variant-2',
      variantName: 'Variant-B',
    },
  ] as Variant[];
  const mockGoals: ExperimentGoal[] = [
    {
      id: 'goal-1',
      name: 'Goal 1',
      type: 'PAGE',
      readOnly: false,
      targetPage: '/target-page',
      mountId: 'some-mount-id',
    },
    {
      id: 'goal-2',
      name: 'Goal 2',
      type: 'PAGE',
      readOnly: false,
      targetPage: '/target-page',
      mountId: 'some-mount-id',
    },
  ];

  beforeEach(async () => {
    const variantsServiceMock = {
      groupVariants: jest.fn(() => [{ id: 'variant-1', name: 'Variant (2 variants)', numberOfVariants: 2 }]),
    };

    fixture = TestBed.configureTestingModule({
      imports: [
        FormsModule,
        MatFormFieldModule,
        MatSelectModule,
        MatButtonModule,
        MatDividerModule,
        NoopAnimationsModule,
      ],
      declarations: [
        ExperimentStartFormComponent,
        TranslateMockPipe,
      ],
      providers: [
        { provide: VariantsService, useValue: variantsServiceMock },
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(ExperimentStartFormComponent);

    component = fixture.componentInstance;

    component.variants = mockVariants;
    component.goals = mockGoals;

    component.ngOnChanges({ variants: new SimpleChange(undefined, component.variants, true) });

    fixture.detectChanges();
  });

  it('should be shown', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should be shown in the enabled mode', () => {
    component.disabled = true;

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should be shown in the disabled mode', () => {
    component.disabled = true;

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should have the first variant preselected', async () => {
    expect(component.selectedVariant).toBe(component.groupedVariants[0]);
  });

  it('should return the original variant', () => {
    const firstOriginalVariant = component.variants[0];

    expect(component.getOriginalVariant(component.groupedVariants[0])).toEqual(firstOriginalVariant);
  });

  describe('if a goal is selected', () => {
    beforeEach(() => {
      component.selectedGoalId = 'goal-1';

      fixture.detectChanges();
    });

    it('should be shown', () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });

    it('should emit the output', () => {
      let result!: SelectedVariantIdAndGoalId;

      component.selected.pipe(first()).subscribe(x => result = x);

      component.onSave();

      expect(result).toEqual({
        variantId: 'variant-1',
        goalId: 'goal-1',
      });
    });
  });
});
