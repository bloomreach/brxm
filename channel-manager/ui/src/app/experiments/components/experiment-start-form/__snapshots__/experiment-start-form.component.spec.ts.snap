// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentStartFormComponent if a goal is selected should be shown 1`] = `
<div
  id="root5"
>
  
  
  
  
  
  <em-experiment-info-bar>
     EXPERIMENT_INFO_START_ALL_VARIANTS_GOAL_REQUIRED 
  </em-experiment-info-bar>
  
  <div
    class="form-wrapper"
  >
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label _mat-animation-noopable mat-form-field-should-float"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="false"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-variants-select mat-select"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  
                  <span
                    class="mat-select-value-text"
                  >
                    
                    <span>
                      Variant (2 variants)
                    </span>
                    
                  </span>
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label"
              >
                
                
                <mat-label
                  class="qa-variant-select-label"
                >
                   VARIANTS 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label _mat-animation-noopable mat-form-field-should-float"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="false"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-goal-select mat-select"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  
                  <span
                    class="mat-select-value-text"
                  >
                    
                    <span>
                      Goal 1
                    </span>
                    
                  </span>
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label"
              >
                
                
                <mat-label
                  class="qa-goal-select-label"
                >
                   EXPERIMENT_GOAL 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </div>
  <mat-divider
    aria-orientation="horizontal"
    class="mat-divider mat-divider-horizontal"
    role="separator"
  />
  <mat-toolbar>
    <button
      class="qa-discard mat-focus-indicator mat-stroked-button mat-button-base _mat-animation-noopable"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         DISCARD 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="save-button qa-save mat-focus-indicator mat-stroked-button mat-button-base mat-primary _mat-animation-noopable"
      color="primary"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         SAVE 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-toolbar>
</div>
`;

exports[`ExperimentStartFormComponent should be shown 1`] = `
<div
  id="root0"
>
  
  
  
  
  
  <em-experiment-info-bar>
     EXPERIMENT_INFO_START_ALL_VARIANTS_GOAL_REQUIRED 
  </em-experiment-info-bar>
  
  <div
    class="form-wrapper"
  >
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="false"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-variants-select mat-select mat-select-empty"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  <span
                    class="mat-select-placeholder"
                  >
                    Â 
                  </span>
                  
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label mat-empty mat-form-field-empty"
              >
                
                
                <mat-label
                  class="qa-variant-select-label"
                >
                   VARIANTS 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="false"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-goal-select mat-select mat-select-empty"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  <span
                    class="mat-select-placeholder"
                  >
                    Â 
                  </span>
                  
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label mat-empty mat-form-field-empty"
              >
                
                
                <mat-label
                  class="qa-goal-select-label"
                >
                   EXPERIMENT_GOAL 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </div>
  <mat-divider
    aria-orientation="horizontal"
    class="mat-divider mat-divider-horizontal"
    role="separator"
  />
  <mat-toolbar>
    <button
      class="qa-discard mat-focus-indicator mat-stroked-button mat-button-base _mat-animation-noopable mat-button-disabled"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         DISCARD 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="save-button qa-save mat-focus-indicator mat-stroked-button mat-button-base mat-primary _mat-animation-noopable mat-button-disabled"
      color="primary"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         SAVE 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-toolbar>
</div>
`;

exports[`ExperimentStartFormComponent should be shown in the disabled mode 1`] = `
<div
  id="root2"
>
  
  
  
  
  
  <em-experiment-info-bar>
     EXPERIMENT_INFO_START_ALL_VARIANTS_GOAL_REQUIRED 
  </em-experiment-info-bar>
  
  <div
    class="form-wrapper"
  >
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label _mat-animation-noopable mat-form-field-should-float mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="true"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-variants-select mat-select mat-select-disabled"
              role="listbox"
              tabindex="-1"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  
                  <span
                    class="mat-select-value-text"
                  >
                    
                    <span>
                      Variant (2 variants)
                    </span>
                    
                  </span>
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label"
              >
                
                
                <mat-label
                  class="qa-variant-select-label"
                >
                   VARIANTS 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="true"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-goal-select mat-select mat-select-empty mat-select-disabled"
              role="listbox"
              tabindex="-1"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  <span
                    class="mat-select-placeholder"
                  >
                    Â 
                  </span>
                  
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label mat-empty mat-form-field-empty"
              >
                
                
                <mat-label
                  class="qa-goal-select-label"
                >
                   EXPERIMENT_GOAL 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </div>
  <mat-divider
    aria-orientation="horizontal"
    class="mat-divider mat-divider-horizontal"
    role="separator"
  />
  <mat-toolbar>
    <button
      class="qa-discard mat-focus-indicator mat-stroked-button mat-button-base _mat-animation-noopable mat-button-disabled"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         DISCARD 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="save-button qa-save mat-focus-indicator mat-stroked-button mat-button-base mat-primary _mat-animation-noopable mat-button-disabled"
      color="primary"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         SAVE 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-toolbar>
</div>
`;

exports[`ExperimentStartFormComponent should be shown in the enabled mode 1`] = `
<div
  id="root1"
>
  
  
  
  
  
  <em-experiment-info-bar>
     EXPERIMENT_INFO_START_ALL_VARIANTS_GOAL_REQUIRED 
  </em-experiment-info-bar>
  
  <div
    class="form-wrapper"
  >
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label _mat-animation-noopable mat-form-field-should-float mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="true"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-variants-select mat-select mat-select-disabled"
              role="listbox"
              tabindex="-1"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  
                  <span
                    class="mat-select-value-text"
                  >
                    
                    <span>
                      Variant (2 variants)
                    </span>
                    
                  </span>
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label"
              >
                
                
                <mat-label
                  class="qa-variant-select-label"
                >
                   VARIANTS 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
    <mat-form-field
      appearance="outline"
      class="mat-form-field mat-primary mat-form-field-type-mat-select mat-form-field-appearance-outline mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable mat-form-field-disabled"
    >
      <div
        class="mat-form-field-wrapper"
      >
        <div
          class="mat-form-field-flex"
        >
          
          
          <div
            class="mat-form-field-outline"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          <div
            class="mat-form-field-outline mat-form-field-outline-thick"
          >
            <div
              class="mat-form-field-outline-start"
            />
            <div
              class="mat-form-field-outline-gap"
            />
            <div
              class="mat-form-field-outline-end"
            />
          </div>
          
          <div
            class="mat-form-field-infix"
          >
            <mat-select
              aria-disabled="true"
              aria-invalid="false"
              aria-multiselectable="false"
              aria-required="false"
              class="qa-goal-select mat-select mat-select-empty mat-select-disabled"
              role="listbox"
              tabindex="-1"
            >
              <div
                aria-hidden="true"
                cdk-overlay-origin=""
                class="mat-select-trigger"
              >
                <div
                  class="mat-select-value"
                >
                  
                  <span
                    class="mat-select-placeholder"
                  >
                    Â 
                  </span>
                  
                </div>
                <div
                  class="mat-select-arrow-wrapper"
                >
                  <div
                    class="mat-select-arrow"
                  />
                </div>
              </div>
              
            </mat-select>
            <span
              class="mat-form-field-label-wrapper"
            >
              
              <label
                class="mat-form-field-label mat-empty mat-form-field-empty"
              >
                
                
                <mat-label
                  class="qa-goal-select-label"
                >
                   EXPERIMENT_GOAL 
                </mat-label>
                
                
              </label>
            </span>
          </div>
          
        </div>
        
        <div
          class="mat-form-field-subscript-wrapper"
        >
          
          
          <div
            class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
            style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
          >
            
            <div
              class="mat-form-field-hint-spacer"
            />
          </div>
        </div>
      </div>
    </mat-form-field>
  </div>
  <mat-divider
    aria-orientation="horizontal"
    class="mat-divider mat-divider-horizontal"
    role="separator"
  />
  <mat-toolbar>
    <button
      class="qa-discard mat-focus-indicator mat-stroked-button mat-button-base _mat-animation-noopable mat-button-disabled"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         DISCARD 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="save-button qa-save mat-focus-indicator mat-stroked-button mat-button-base mat-primary _mat-animation-noopable mat-button-disabled"
      color="primary"
      disabled="true"
      mat-stroked-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         SAVE 
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-toolbar>
</div>
`;
